pageserver/src/tenant/timeline.rs

```diff
@@ -1175,22 +1179,34 @@ impl Timeline {
         let reconstruct_timer = crate::metrics::RECONSTRUCT_TIME
             .for_get_kind(get_kind)
             .start_timer();
-        let mut results: BTreeMap<Key, Result<Bytes, PageReconstructError>> = BTreeMap::new();
         let layers_visited = reconstruct_state.get_layers_visited();
 
-        for (key, res) in std::mem::take(&mut reconstruct_state.keys) {
-            match res {
-                Err(err) => {
-                    results.insert(key, Err(err));
-                }
-                Ok(state) => {
-                    let state = ValueReconstructState::from(state);
+        let futs = FuturesUnordered::new();
```

Comment:
Above, we bail if `get_vectored_reconstruct_data` returns an error while there are still IOs in flight. Need to fix this.

Reply:
c4e0ba38b8b89807b177381803b0195c9d107aba